### PR TITLE
fix(abstractions): preserve falsy scalar request bodies

### PR DESCRIPTION
**Pull request by rksharma-owg**
`RequestInformation.setContentFromScalar` serialized valid `false`, `0`, and `""` request bodies as `null`. The public request-builder proxy also rejected these values before they could reach the serializer. Both paths used truthiness checks.

Preserve these scalar values in both paths. Keep direct `null`, `undefined`, and `NaN` scalar inputs on the null-writing path, preserving their pre-PR behavior. The proxy still rejects null or undefined required bodies.

Tests cover scalar writer arguments and request content, null/undefined/NaN serialization, proxy request construction and sending, and missing proxy bodies. No public signatures or dependencies change. This does not change the serialization writers' handling of infinities or primitive collections.

### Validation

- New review regression tests on the previous PR commit: 4 failed, 38 passed (three falsy proxy cases and NaN). With the update: 42 passed.
- Independent checks with the real JSON serialization writer: 16 passed across direct and proxy paths.
- `npm run build`, `npm run build:integrated`, `npm run lint:eslint:loud`, and the build's `prettier:check`: passed.
- `npm test`: both configured phases passed, each reporting 397 passed and 1 skipped. The browser-named script is not evidence of actual browser coverage: the earlier environment probe found it running under Node with the current Vitest configuration.
- `npx lerna run test:coverage --scope @microsoft/kiota-abstractions`: 104 passed, 1 skipped.

Validated locally on Node 22.23.1 / macOS arm64. The current upstream commit's hosted build and coverage matrix is green. The original scalar regression also failed on unchanged upstream before the initial fix (3 failed, 25 passed).

Found during runtime boundary testing; no existing issue is linked. The merged multipart-content fix in #2132 concerns a separate code path.

### CODE REVIEW

File: packages/abstractions/test/common/requestInformation.ts [hunk 1]
```diff
@@ -232,29 +232,36 @@ describe("RequestInformation", () => {
 		assert.equal(requestInformation.URL, expected);
 	});
 
-	it("Sets a scalar content", () => {
+	it.each([false, 0, "", true, 1, "some content"])("Sets scalar content without changing %j", (value) => {
```
Comment (baywet) [inline packages/abstractions/test/common/requestInformation.ts:235]: Can we also add a unit test for null/undefined to prevent regressions on that path please?

Comment (rksharma-owg) [reply packages/abstractions/test/common/requestInformation.ts:235]: Added in 86cb427: null and undefined regression cases assert that writeNullValue is called and the request content is null. These pass alongside the scalar-value tests.

File: packages/abstractions/src/requestInformation.ts [hunk 2]
```diff
@@ -198,7 +198,7 @@ export class RequestInformation implements RequestInformationSetContent {
 				} else {
 					const valueType = typeof value;
 					span.setAttribute(RequestInformation.requestTypeKey, valueType);
-					if (!value) {
+					if (value === null || value === undefined) {
```
[the remark this answers is not in the document]
Comment (rksharma-owg) [reply packages/abstractions/src/requestInformation.ts]: Updated in 86cb427. The proxy guard now rejects only null/undefined, with regression cases through both toPostRequestInformation and post for false, 0, and the empty string, plus truthy controls and missing-body rejection. I also preserved the pre-PR scalar NaN-to-null behavior and added coverage for it. The new tests failed in the four reported cases before the update and now pass; 16 additional assertions using the real JSON writer pass too.

File: packages/abstractions/src/requestInformation.ts [hunk 3]
```diff
@@ -198,7 +198,7 @@ export class RequestInformation implements RequestInformationSetContent {
 				} else {
 					const valueType = typeof value;
 					span.setAttribute(RequestInformation.requestTypeKey, valueType);
-					if (!value) {
+					if (value === null || value === undefined || Number.isNaN(value)) {
```
Comment (baywet) [inline packages/abstractions/src/requestInformation.ts:201]: Isn't the is nan fail on object types like date only etc?

Comment (rksharma-owg) [reply packages/abstractions/src/requestInformation.ts:201]: Thanks for checking this. This uses `Number.isNaN`, which does not coerce its argument, unlike the global `isNaN`. It returns false for these objects, so they still reach their existing `instanceof` branches.

I verified `Date`, `DateOnly`, `TimeOnly`, and `Duration` through `setContentFromScalar` with the real JSON writer on 86cb427; all four serialize correctly. For example, `new DateOnly({ year: 2026, month: 9, day: 13 })` returns false from `Number.isNaN` and serializes as `"2026-09-13"`.

**Review by baywet — APPROVED** [review]
Thank you for making the changes!

**Comment by rksharma-owg** [comment]
Follow-up on the CI notifications: the two red runs were approval-held workflows for the superseded first commit `c791349`. They contain no jobs and no logs, so no test executed or failed. GitHub finalized those stale runs when this PR merged:

- [Build and test run 34734709778](https://github.com/microsoft/kiota-typescript/actions/runs/34734709778)
- [Dependabot auto-merge run 34734709799](https://github.com/microsoft/kiota-typescript/actions/runs/34734709799)

I revalidated the final PR head `86cb427a724df737af0081cb64f5b25425db6c50`:

- Local Node 24.21.0/macOS arm64: `npm ci`, `npm run build`, `npm run lint:eslint:loud`, `npm run prettier:check`, `npm run build:integrated`, `npm test`, and `npm run test:coverage` all passed. Both test phases reported 397 passed and 1 skipped; coverage reported the same totals.
- [Fork preflight run 34970266050](https://github.com/rksharma-owg/kiota-typescript/actions/runs/34970266050): all 3 Ubuntu build/test jobs (Node 20, 22, and 24), all 9 package coverage jobs, and the matrix gate passed. Every runner verified the exact commit before testing.
- [Final upstream PR-head run 34757491862](https://github.com/microsoft/kiota-typescript/actions/runs/34757491862) passed, and the [post-merge build for `b86c09a`](https://github.com/microsoft/kiota-typescript/actions/runs/34969313027) also passed.

I added a [manual exact-SHA preflight workflow](https://github.com/rksharma-owg/kiota-typescript/blob/main/.github/workflows/fork-preflight.yml) to the fork so future contribution branches can run this matrix before PR submission. No upstream source change is needed for these notifications.
